lib/liquid/condition.rb

```diff
@@ -47,6 +48,11 @@ def evaluate(context = Context.new)
       when :or
         result || @child_condition.evaluate(context)
       when :and
+        @@depth += 1
+        if @@depth >= 500
+          @@depth = 0
```

Comment:
This is the only place where the depth is reduced, which means that we would get an error every 500 times an liquid `if` condition is evaluated.  Instead the depth needs to be incremented and decremented around the rendering a block.